src/js/relayclient/ServerHelper.js

```diff
@@ -0,0 +1,168 @@
+
+
+const addPastEvents = require('./addPastEvents')
+
+class ActiveRelayPinger {
+
+    // TODO: 'httpSend' should be on a network layer
+    constructor(filteredRelays, httpSend) {
+        this.filteredRelays = filteredRelays
```

Comment:
you modify the referenced list you received.
instead, you should shallow-copy it, using `array.slice()`

Reply:
Currently, I do not technically modify this.filteredRelay, but re-assign it, but that is a lucky coincidence. Fixed.